securedrop/journalist.py

```diff
@@ -409,6 +411,55 @@ def remove_star(sid):
     return redirect(url_for('index'))
 
 
+@app.route('/profile/<username>')
+@login_required
+def profile(username):
+    user = Journalist.query.filter_by(username=username).one()
+    sources = Source.query.filter_by(journalist_id=user.id).all()
+    return render_template("profile.html", user=user, sources=sources)
+
+
+@app.route('/edit-profile/', methods=['GET', 'POST'])
+@login_required
+def edit_profile():
+    form = forms.EditProfileForm(request.form)
+    if request.method == 'GET':
+        # Populate the inputs with the existing values in the database
+        form.full_name.data = g.user.full_name
+        form.about.data = g.user.about
+        form.pgp_key.data = g.user.pgp_key
+    elif request.method == 'POST' and form.validate():
+        user = Journalist.query.filter_by(username=g.user.username).one()
+        user.full_name = form.full_name.data
+        user.about = form.about.data
+        user.pgp_key = form.pgp_key.data
+        db_session.commit()
+        flash('Successfully edited your profile!', 'notification')
+        return redirect(url_for('profile', username=g.user.username))
+    return render_template("edit_profile.html", form=form, user=g.user)
+
+
+@app.route('/admin/edit-profile/<int:user_id>', methods=('GET', 'POST'))
```

Comment:
I think the `/admin/edit-profile/<username>/` URL routing scheme is nicer, but not super important.